plugins/inputs/f5_load_balancer/f5_load_balancer.go

```diff
@@ -0,0 +1,343 @@
+			fields["pool_serverside_packets_in"], _ = strconv.Atoi(gjson.Get(base, "serverside\\.pktsIn.value").Raw)
+			fields["pool_serverside_packets_out"], _ = strconv.Atoi(gjson.Get(base, "serverside\\.pktsOut.value").Raw)
+			fields["pool_serverside_total_connections"], _ = strconv.Atoi(gjson.Get(base, "serverside\\.totConns.value").Raw)
+			fields["pool_total_requests"], _ = strconv.Atoi(gjson.Get(base, "totRequests.value").Raw)
+			available := 0
+			if gjson.Get(base, "status\\.availabilityState.description").Str == "available" {
+				available = 1
+			}
+			fields["pool_available"] = available
+
+			acc.AddFields("f5_load_balancer", fields, tags)
+		}(&wg, url, acc)
+	}
+	wg.Wait()
+}
+
+func (f5 *F5LoadBalancer) GatherVirtual(acc telegraf.Accumulator, parentWG *sync.WaitGroup) {
+	defer parentWG.Done()
+	urls, err := f5.GetUrls("/mgmt/tm/ltm/virtual")
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	var wg sync.WaitGroup
+	for _, url := range urls {
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, url string, acc telegraf.Accumulator) {
+			defer wg.Done()
+			resp, tags, err := f5.GetTags(url)
+			if err != nil {
+				acc.AddError(err)
+				return
+			}
+			base := gjson.Get(resp, "entries.*.nestedStats.entries").Raw
+			fields := make(map[string]interface{})
+			fields["virtual_clientside_bits_in"], _ = strconv.Atoi(gjson.Get(base, "clientside\\.bitsIn.value").Raw)
+			fields["virtual_clientside_bits_out"], _ = strconv.Atoi(gjson.Get(base, "clientside\\.bitsOut.value").Raw)
+			fields["virtual_clientside_current_connections"], _ = strconv.Atoi(gjson.Get(base, "clientside\\.curConns.value").Raw)
+			fields["virtual_clientside_packets_in"], _ = strconv.Atoi(gjson.Get(base, "clientside\\.pktsIn.value").Raw)
+			fields["virtual_clientside_packets_out"], _ = strconv.Atoi(gjson.Get(base, "clientside\\.pktsOut.value").Raw)
+			fields["virtual_total_requests"], _ = strconv.Atoi(gjson.Get(base, "totRequests.value").Raw)
+			fields["virtual_one_minute_avg_usage"], _ = strconv.Atoi(gjson.Get(base, "oneMinAvgUsageRatio.value").Raw)
+			available := 0
+			if gjson.Get(base, "status\\.availabilityState.description").Str == "available" {
+				available = 1
+			}
+			fields["virtual_available"] = available
+
+			acc.AddFields("f5_load_balancer", fields, tags)
+		}(&wg, url, acc)
+	}
+	wg.Wait()
+}
+
+func (f5 *F5LoadBalancer) GatherNetInterface(acc telegraf.Accumulator, parentWG *sync.WaitGroup) {
+	defer parentWG.Done()
+	urls, err := f5.GetUrls("/mgmt/tm/net/interface")
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	var wg sync.WaitGroup
+	for _, url := range urls {
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, url string, acc telegraf.Accumulator) {
+			defer wg.Done()
+			resp, err := f5.Call(url)
+			if err != nil {
+				acc.AddError(err)
+				return
+			}
+			tags := map[string]string{}
+			tags["name"] = gjson.Get(resp, "entries.*.nestedStats.entries.tmName.description").Raw
```

Comment:
Have you considered unmarshalling into a Go struct instead of using gjson? There could be bug if gjson.Get returns a slice in some of these cases where it is gathering using wildcards.

Reply:
Would you be fine with me using gjson if I got rid of the wildcard? The wildcard is there just because the value in that position is a URL that changes based on the resource you're collecting metrics from (see the json response in the test case to see what I mean). I can build that value from another field in the response with a few lines of code in order to get rid of the wildcard.

I'm not a fan of how Go handles json. I originally started down the unmarshalling path but due to how nested the data was, it started to become messy. I found that gjson was already a project dependency and found that path much more tenable.

Reply:
@cthiel42 sorry for taking so long to respond to you, after more thought I think your approach is acceptable. I am concerned it might cause this plugin to be slower then using the unmarhsalling approach, but if you have tested this and its working for you then we should be fine.